src/SFA.DAS.EmployerAccounts.Web/Orchestrators/SearchPensionRegulatorOrchestrator.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EmployerAccounts.Extensions;
+using SFA.DAS.EmployerAccounts.Interfaces;
+using SFA.DAS.EmployerAccounts.Models.Account;
+using SFA.DAS.EmployerAccounts.Models.PensionRegulator;
+using SFA.DAS.EmployerAccounts.Queries.GetPensionRegulator;
+using SFA.DAS.EmployerAccounts.Web.ViewModels;
+using SFA.DAS.Validation;
+
+namespace SFA.DAS.EmployerAccounts.Web.Orchestrators
+{
+    public class SearchPensionRegulatorOrchestrator : UserVerificationOrchestratorBase 
+    {
+        private readonly ICookieStorageService<EmployerAccountData> _cookieService;
+        private const string CookieName = "sfa-das-employerapprenticeshipsservice-employeraccount";
+
+        protected SearchPensionRegulatorOrchestrator()
+        {
+
+        }
+
+        public SearchPensionRegulatorOrchestrator(IMediator mediator, ICookieStorageService<EmployerAccountData> cookieService)
+            : base(mediator)
+        {
+            _cookieService = cookieService;
+        }
+
+        public virtual async Task<OrchestratorResponse<SearchPensionRegulatorResultsViewModel>> SearchPensionRegulator(string payeRef)
+        {
+            var response = new OrchestratorResponse<SearchPensionRegulatorResultsViewModel>();
+
+            try
+            {
+                var result = await Mediator.SendAsync(new GetPensionRegulatorRequest {PayeRef = payeRef});
+                response.Data = new SearchPensionRegulatorResultsViewModel
+                {
+                    Results = CreateResult(result.Organisations).ToList(),                    
+                    PayeRef = payeRef
+                };
+            }
+            catch (InvalidRequestException ex)
+            {
+                response.Exception = ex;
+                response.FlashMessage = FlashMessageViewModel.CreateErrorFlashMessageViewModel(ex.ErrorMessages);
+                response.Status = HttpStatusCode.BadRequest;
+            }
+            catch (Exception)
+            {
+                response = new OrchestratorResponse<SearchPensionRegulatorResultsViewModel>
```

Comment:
This looks a bit odd; should be not, at least log the exception? We are saying if we get `InvalidRequestException` we deal with that and show the user an error message. Any other exception we just swallow and do nothing?

Reply:
Agree. I've added logging into the exception and removed the InvalidRequestException catch. The requirement is that if the PR search fails then it swallows the error and goes to the organiusation search.